### PR TITLE
html2parquet feature added

### PR DESCRIPTION
**Pull request by sungeunan-ibm**
## Why are these changes needed?

Added transform to convert html files in a zip to parquet.

## Related issue number (if any).

None

### CODE REVIEW

File: transforms/universal/html2parquet/python/pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+name = "dpk_noop_transform_python"
+version = "0.2.1.dev0"
+requires-python = ">=3.10"
+description = "NOOP Python Transform"
+license = {text = "Apache-2.0"}
+readme = {file = "README.md", content-type = "text/markdown"}
+authors = [
+    { name = "David Wood", email = "dawood@us.ibm.com" },
+    { name = "Boris Lublinsky", email = "blublinsky@ibm.com" },
+]
```
Comment (blublinsky) [inline transforms/universal/html2parquet/python/pyproject.toml:11]: This has to be the actual authors, not me or David

File: transforms/universal/html2parquet/python/pyproject.toml [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+dependencies = [
+    "data-prep-toolkit==0.2.1.dev0",
+]
```
Comment (blublinsky) [inline transforms/universal/html2parquet/python/pyproject.toml:14]: Pls add here the packages you need. trafilatura, anything else

File: transforms/universal/html2parquet/python/src/html2parquet_main.py [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+class HtmlToParquetTransform(AbstractTableTransform):
+    def __init__(self, config: dict[str, Any]):
+        self.sleep = config.get("sleep", 1)
+
+    def transform_binary(self, file_name: str, byte_array: bytes) -> tuple[list[tuple[bytes, str]], dict[str, Any]]:
+        """
+        Converts raw data file (ZIP) to Parquet format
+        """
+        # We currently only process .zip files
+        print("transform_binary")
```
Comment (blublinsky) [inline transforms/universal/html2parquet/python/src/html2parquet_main.py]: please switch to logger everywhere. no print

File: transforms/universal/html2parquet/python/src/html2parquet_main.py [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+
+        with zipfile.ZipFile(io.BytesIO(bytes(byte_array))) as opened_zip:
+            # Loop through each file member in the ZIP archive
+            for member in opened_zip.infolist():
+                if not member.is_dir():
+                    with opened_zip.open(member) as file:
+                        try:
+                            # Read the content of the file
+                            content_bytes = file.read()
+                            #use Tra something library
```
Comment (blublinsky) [inline transforms/universal/html2parquet/python/src/html2parquet_main.py]: tra something will not cut it. pls use the actual factory

File: transforms/universal/html2parquet/python/src/html2parquet_main.py [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+                            content_string = trafilatura.extract(content_bytes)
+                            print("content_string:", content_string)
+                            row_data = {
+                                "filename": member.filename,
+                                "contents": content_string
+                            }
```
Comment (blublinsky) [inline transforms/universal/html2parquet/python/src/html2parquet_main.py]: Are these the only 2 columns that we need? Pls take a look at pdf2parquet and code2parquet for ideas on additional columns

File: transforms/universal/html2parquet/python/src/html2parquet_main.py [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+                            data.append(row_data)
+                            number_of_rows += 1
+                        except Exception as e:
+                            self.logger.warning(f"Exception {str(e)} processing file {member.filename}, skipping")
+        table = pa.Table.from_pylist(data)
+        return [(TransformUtils.convert_arrow_to_binary(table=table), ".parquet")], {"number of rows": number_of_rows}
+
+
+ 
+    def transform(self, table: pa.Table, file_name: str = None) -> tuple[list[pa.Table], dict[str, Any]]:
```
Comment (blublinsky) [inline transforms/universal/html2parquet/python/src/html2parquet_main.py]: This seems to be a leftover from noop. pls remove

File: transforms/universal/html2parquet/python/src/html2parquet_main.py [hunk 7]
```diff
@@ -0,0 +1,110 @@
...
+# )
+# import data_processing
+# print(dir(data_processing))
+
+
+from data_processing.transform import AbstractTableTransform, AbstractBinaryTransform, TransformConfiguration
+from data_processing.utils import CLIArgumentProvider, get_logger, TransformUtils
+
+
+class HtmlToParquetTransform(AbstractTableTransform):
```
Comment (blublinsky) [inline transforms/universal/html2parquet/python/src/html2parquet_main.py]: You are not doing table transform. plas replace AbstractTableTransform with AbstractBinaryTransform

File: transforms/universal/html2parquet/python/src/html2parquet_main.py [hunk 8]
```diff
@@ -0,0 +1,110 @@
...
+    def __init__(self, config: dict[str, Any]):
+        self.sleep = config.get("sleep", 1)
```
Comment (blublinsky) [inline transforms/universal/html2parquet/python/src/html2parquet_main.py]: This is from Noop. pls remove

File: transforms/universal/html2parquet/python/src/html2parquet_main.py [hunk 9]
```diff
@@ -0,0 +1,110 @@
...
+            type=int,
+            default=1,
+            help="Sleep actor for a number of seconds while processing the data frame, before writing the file to COS",
+        )
+        parser.add_argument(
+            f"--{pwd_cli_param}",
+            type=str,
+            default="nothing",
+            help="A dummy password which should be filtered out of the metadata",
+        )
```
Comment (blublinsky) [inline transforms/universal/html2parquet/python/src/html2parquet_main.py]: The above is from Noop and does not belong here. Pls add only the parameters that you need

File: transforms/universal/html2parquet/python/src/html2parquet_main.py [hunk 10]
```diff
@@ -0,0 +1,110 @@
...
+
+    def apply_input_params(self, args: Namespace) -> bool:
+        captured = CLIArgumentProvider.capture_parameters(args, cli_prefix, False)
+        if captured.get(sleep_key) < 0:
+            print(f"Parameter html2parquet_sleep_sec should be non-negative. you specified {args.noop_sleep_sec}")
+            return False
+        self.params = captured
+        return True
```
Comment (blublinsky) [inline transforms/universal/html2parquet/python/src/html2parquet_main.py]: This has nothing to do with your transform

**Comment by blublinsky** [comment]
please look at the, for example, at pdf2parquet to see a proper layout of the src directory

**Review by daw3rd — CHANGES_REQUESTED** [review]
Please see the [transform project conventions](https://github.com/IBM/data-prep-kit/tree/dev/transforms#transform-project-conventions) and adjust the file structure accordingly.  Thanks.

**Pull request closed without merging**